test/test_action.py

```diff
@@ -33,56 +34,98 @@ def test_parse_directory_configs():
+    notice_file_path = os.path.join(repo_root, notice_file_name)
     # init repo
-    repo = Repo.init("./testbench/git")
+    repo = Repo.init(repo_root)
+    subprocess.run(
+        [
+            "git",
+            "config",
+            "--global",
+            "--add",
+            "safe.directory",
+            repo_root,
+        ]
+    )
 
     # create file
     with open(notice_file_path, "w+", encoding="utf-8") as notice_file:
         notice_file.write("first license")
 
     # add file initially
-    repo.index.add("my-notice-file.md")
+    repo.index.add(notice_file_name)
     repo.index.commit("Initial")
 
-    commit_notice_file_if_dirty("./testbench/git/", "my-notice-file")
     ref_log = repo.active_branch.log()
 
     try:
+        assert not is_dirty(repo_root, notice_file_name)
         assert len(ref_log) == 1
         assert ref_log[0].message == "commit (initial): Initial"
     finally:
         os.remove(notice_file_path)
-        shutil.rmtree("./testbench/git/.git")
+        shutil.rmtree(os.path.join(repo_root, ".git"))
+        subprocess.run(
+            [
+                "git",
+                "config",
+                "--global",
+                "--add",
+                "safe.directory",
+                ROOT_DIR,
+            ]
+        )
 
 
 def test_commit_if_dirty():
     """Test commit is created if file is dirty"""
-    notice_file_path = "./testbench/git/my-notice-file.md"
+    repo_root = os.path.join(ROOT_DIR, "testbench/git")
+    notice_file_name = "my-notice-file.md"
+    notice_file_path = os.path.join(repo_root, notice_file_name)
 
     # init repo
-    repo = Repo.init("./testbench/git")
+    repo = Repo.init(repo_root)
+    subprocess.run(
+        [
+            "git",
+            "config",
+            "--global",
+            "--add",
+            "safe.directory",
+            repo_root,
+        ]
+    )
 
     # create file
     with open(notice_file_path, "w+", encoding="utf-8") as notice_file:
         notice_file.write("first license")
 
     # add file initially
-    repo.index.add("my-notice-file.md")
+    repo.index.add([notice_file_name])
     repo.index.commit("Initial")
 
     # modify file
-    with open(notice_file_path, "w+", encoding="utf-8") as notice_file:
+    with open(notice_file_path, "a", encoding="utf-8") as notice_file:
         notice_file.write("new license")
 
-    commit_notice_file_if_dirty("./testbench/git/", "my-notice-file", push=False)
     ref_log = repo.active_branch.log()
 
     try:
-        assert len(ref_log) == 2
+        assert is_dirty(repo_root, notice_file_name)
+        assert len(ref_log) == 1
         assert ref_log[0].message == "commit (initial): Initial"
-        assert ref_log[1].message == "Update my-notice-file.md"
     finally:
         os.remove(notice_file_path)
-        shutil.rmtree("./testbench/git/.git")
+        shutil.rmtree(os.path.join(repo_root, ".git"))
+        subprocess.run(
```

Comment:
I cannot figure out why you need this call?

Reply:
in the test the safe directory is set to the newly created repo, this is a cleanup step to make the real repository usable again